src/core/config.go

```diff
@@ -296,12 +297,25 @@ func ReadConfigFiles(filenames []string, profiles []string) (*Configuration, err
 	})
 }
 
-// normalisePluginConfigKeys converts all config for plugins to lower case
-func normalisePluginConfigKeys(config *Configuration) {
+// normaliseAndValidatePluginConfigKeys converts all config for plugins to lower case, and
+// validates against config fields provided by the plugin
+func normaliseAndValidatePluginConfigKeys(config *Configuration) {
 	for _, plugin := range config.Plugin {
 		newExtraValues := make(map[string][]string, len(plugin.ExtraValues))
 		for k, v := range plugin.ExtraValues {
 			newExtraValues[strings.ToLower(k)] = v
+			// Can only validate if there's anything to check against. The plugin config
+			// hasn't necessarily been loaded yet.
+			valid := len(config.PluginConfig) == 0
+			for _, val := range config.PluginConfig {
+				if strings.EqualFold(k, val.ConfigKey) {
+					valid = true
+					break
+				}
+			}
```

Comment:
This logic is being applied to all `config.Plugin`s and I reckon this will cause some issues, as the PluginConfig/PluginDefinition is specific to one single plugin. For instance, if the PluginConfig is related to Python, but you are going over the Java plugin then the config keys of Java won't be found on the PluginConfig for Python

Reply:
Yeah I did think that but it should be fine, right? Because it will eventually loop over the Java config and find something. As in, it always loops over all the config for every plugin you have, so it will eventually find one that matches. At least, I think that's how it works. But anyway even if it does work, it's not super pretty. I'll see if I can improve the logic here.

Reply:
As discussed, you're right it doesn't work the way I thought. I'll get on the fix!